src/main/kotlin/org/jmailen/gradle/kotlinter/support/EditorConfigUtils.kt

```diff
@@ -12,20 +15,39 @@ internal fun editorConfigOverride(ktLintParams: KtLintParams): EditorConfigOverr
     val rules = ktLintParams.disabledRules
 
     return if (rules.isEmpty()) {
-        EditorConfigOverride.emptyEditorConfigOverride
+        EMPTY_EDITOR_CONFIG_OVERRIDE
     } else {
-        EditorConfigOverride.from(DefaultEditorConfigProperties.ktlintDisabledRulesProperty to rules.joinToString(separator = ","))
+        rules
```

Comment:
🔧 
My only suggestion here would be to add additional set of functional tests to ensure most common scenarios 
i.e. experimental rules can also be disabled. I don't think we have that covered anywhere 👀

Reply:
Well, good catch: I added a test for disabling an experimental rule (which is not in the _standard_ ruleset) and it passed, but at the same time I added a positive test case to check that the experimental rule is executed in the first place if it's not disabled — it's not. This seems to be an unintended [behavior change](https://github.com/pinterest/ktlint/issues/1747) in ktlint 0.48.0, I assume it's fixed in 0.48.1. For now the PR will fail tests, hopefully after updating ktlint dependency to 0.48.1 it will pass.

As for using experimental rules: I understand it's not ideal and the tests would have to change as experimental rules evolve, but I didn't see an easy way to use a test-only custom rule set in the functional tests. If you have a suggestion on how to do that, I can try

Reply:
0.48.1 is out now, so we can bump this PR to that perhaps?

Reply:
I did that, but seems like either some artifacts are still not available or the CI is flaky — 16/21 checks passed, and quick look at the logs shows stuff I'm not sure is related to this PR 🤔

Reply:
Some of the checks failed due to code style offences:
<img width="824" alt="image" src="https://user-images.githubusercontent.com/36954793/211162367-ac94a291-9464-4bb6-a4c6-7a2d6c0d7bc1.png">

the rest are caused by plugin's Gradle 8.0 incompatibility, which I'll try to address in: https://github.com/jeremymailen/kotlinter-gradle/pull/306

Reply:
#306 is merged now, so maybe try rebasing?

Reply:
The tests were still failing, so I reproed and reported here: https://github.com/pinterest/ktlint/issues/1768

Reply:
Okay so, I think I've fixed the extension and I kept old behavior (using `EditorConfigOverride`, but after discussion in https://github.com/pinterest/ktlint/issues/1768 I'm somewhat convinced that Kotlinter should _not_ offer API to enable/disable rules, experimental or otherwise. And if it's really desireable, it shouldn't use `EditorConfigOverride` for that but rather `EditorConfigDefaults`.

Long explanation is here: https://github.com/paul-dingemans/ktlint/commit/cfecedcac8865d5a2c57466572ed387255e861a6

My take is that Ktlint respects hierarchical `.editorconfig`: you can override rules for each directory, while Kotlinter offers just a single global setting. The only reasonable behavior for Kotlinter is to provide its settings as a default rather than override, otherwise it's disallowing per-directory settings. But there are arguments to be made for removing Kotlinter API for configuring Ktlint completely:
 - users should have `.editorconfig` file in their project root directory even if they don't have any settings there, because without `root = true` Ktlint will search for `.editorconfig` outside of the project root directory (last I heard)
 - once there exists an  `.editorconfig` file, it's reasonable to have it as a central source of configuration. Having global defaults/overrides in Kotlinter seems like a small benefit. Perhaps the API was added when it was more difficult (or impossible) to use `.editorconfig`, but clearly Ktlint considers it a sole source of configuration now and I think it's best not to fight it

Reply:
I fully agree 👍  I'd strongly suggest moving away from `EditorConfigOverride` in favour of `EditorConfigDefaults`, as [Paul suggested ](https://github.com/pinterest/ktlint/issues/1768#issuecomment-1383196514)- it should be less confusing for the user.

Actually, I'd even consider taking a step further, and deprecate both `disabled_rules` and `experimentalRules` within the plugin, since it probably should be preferred to pass them via `.editorconfig` properties (wouldn't it?). My main motivation here is to get rid of some of the complexity we're adding, just to preserve backwards compatibility. Moreover, AFAICS configuring rules via `.editorconfig` requires that same amount of effort as via Plugin's extension, so the extension doesn't provide that much of a value anymore.
This might require a separate discussion though 👀

Reply:
Yes, I agree. We should stop configuring rules altogether in our gradle config block. The config block remains but just for kotlinter specific gradle configurations such as ignore failures and reporters. A few consequences:

- Our next kotlinter release will be 4.x compatibility breaking
- We can include the other compatibility breaking changes in this release #279 and #294 
- Sounds like we need not wait for ktlint 0.48.2 since we don't need that editorconfig fix anymore?

Reply:
> Sounds like we need not wait for ktlint 0.48.2 since we don't need that editorconfig fix anymore?

Looks like it to me 🙂 Even with current implementation we don't need 0.48.2 I think, only 0.48.0 was breaking Kotlinter

@jeremymailen Please let me know how you'd like to proceed with this PR. Personally I'd prefer to merge it as is, as it's been open for a while now. Removing/deprecating rules configuration will probably involve writing some docs so I'd prefer to leave that to you (or at least a separate review)